packages/kbn-apm-synthtrace/src/lib/entity/entity_syntrace_es_client.ts

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { Client } from '@elastic/elasticsearch';
+import { PassThrough, Readable, Transform, pipeline } from 'stream';
+import { ESDocumentWithOperation } from '@kbn/apm-synthtrace-client';
+import moment from 'moment';
+import fetch from 'node-fetch';
+import { SynthtraceEsClient } from '../shared/base_client';
+import { Logger } from '../utils/create_logger';
+import { getSerializeTransform } from '../shared/get_serialize_transform';
+import { getDedotTransform } from '../shared/get_dedot_transform';
+import { createEntityLatestAggregator } from './aggregators/create_entity_aggregator';
+import { fork } from '../utils/stream_utils';
+import { kibanaHeaders } from '../shared/client_headers';
+import { getFetchAgent } from '../../cli/utils/ssl';
+
+interface EntityDefinitionResponse {
+  definitions: Array<{ type: string; state: { installed: boolean; running: boolean } }>;
+}
+
+export class EntitySynthtraceEsClient extends SynthtraceEsClient<any> {
+  private kibanaTarget: string;
+
+  constructor({
+    kibanaTarget,
+    ...options
+  }: { client: Client; logger: Logger; kibanaTarget: string } & any) {
+    super({
+      ...options,
+      pipeline: entityPipeline(),
+    });
+    this.kibanaTarget = kibanaTarget;
+    this.indices = [
+      '.entities.v1.history.builtin_services*',
+      '.entities.v1.latest.builtin_services*',
+    ];
+  }
+
+  async installEntityIndexPatterns() {
+    const url = `${this.kibanaTarget}/internal/entities/definition`;
+    const response = await fetch(url, {
+      method: 'GET',
+      headers: kibanaHeaders(),
+      agent: getFetchAgent(url),
+    });
+    const entityDefinition: EntityDefinitionResponse = await response.json();
+
+    const hasServiceEntityDefinition = entityDefinition.definitions.find(
+      (definition) => definition.type === 'service'
+    )?.state.installed;
```

Comment:
We changed `GET /definition` in https://github.com/elastic/kibana/pull/191933. `state.installed` was previously included by default in the response but it now needs to be explicitly requested with `?includeState=true`